bin/license.py

```diff
@@ -0,0 +1,387 @@
+      the Work or Derivative Works thereof, You may choose to offer,
+      and charge a fee for, acceptance of support, warranty, indemnity,
+      or other liability obligations and/or rights consistent with this
+      License. However, in accepting such obligations, You may act only
+      on Your own behalf and on Your sole responsibility, not on behalf
+      of any other Contributor, and only if You agree to indemnify,
+      defend, and hold each Contributor harmless for any liability
+      incurred by, or claims asserted against, such Contributor by reason
+      of your accepting any such warranty or additional liability.
+
+   END OF TERMS AND CONDITIONS
+
+   APPENDIX: How to apply the Apache License to your work.
+
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "{}"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright {yyyy} {name of copyright owner}
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+
+=======================================================================
+Apache SeaTunnel (Incubating) Subcomponents:
+
+The Apache SeaTunnel(Incubating) project contains subcomponents with separate copyright
+notices and license terms. Your use of the source code for the these
+subcomponents is subject to the terms and conditions of the following
+licenses.
+
+
+
+"""
+
+for k, v in licenses_map.items():
+    if len(v) == 0:
+        continue
+    res += "========================================================================\n"
+    res += k
+    res += '\n'
+    res += "========================================================================\n\n"
+    res += licenses_describe_map[k]
+    res += '\n'
+    for _ in sorted(v):
+        res += "     "
+        res += _
+        res += '\n'
+    res += '\n\n'
+
+if print_diff == 'true':
+    tmp_file = third_party + ".tmp"
+    with open(tmp_file, "w") as f:
+        f.write(res)
+    print("Please modify the LICENSE file according to the diff information.")
+    diff_res = os.system("diff " + license + " " + tmp_file)
```

Comment:
This is not portable, Windows system won't run this command `diff`, IIRC

Reply:
It's done, now it supports Windows